pkg/shell/superuser.tsx

```diff
@@ -38,7 +36,9 @@ import { LockIcon } from '@patternfly/react-icons';
 
 const _ = cockpit.gettext;
 
-function sudo_polish(msg) {
+function sudo_polish(msg: string): string;
+function sudo_polish(msg: null): null;
+function sudo_polish(msg: string | null) {
```

Comment:
Best we can do is this but as you said in Matrix we don't get the any type checking in the function to make sure the overloads match..
```suggestion
function sudo_polish(msg: string): string;
function sudo_polish(msg: null): null;
function sudo_polish(msg: string | null): string | null {
```